libraries/ssh_crypto.rb

```diff
@@ -94,21 +99,25 @@ def valid_kexs # rubocop:disable Metrics/CyclomaticComplexity
       when /^7\./, /^8\./
         kex = kex66
       end
-    when 'amazon', 'fedora', 'alpine'
+    # https://pkgs.alpinelinux.org/packages?name=openssh
+    when 'alpine'
+      kex = inspec.os[:release].split('.')[1] >= '10' ? kex80 : kex66
+    when 'amazon'
       kex = kex66
+    # https://src.fedoraproject.org/rpms/openssh
+    when 'fedora'
+      kex = inspec.os[:release] >= '30' ? kex80 : kex66
+    # https://software.opensuse.org/package/openssh
     when 'opensuse'
-      case inspec.os[:release]
-      when /^13\.2/
-        kex = kex66
-      when /^42\./
-        kex = kex66
-      end
+      kex = inspec.os[:release] == '15.2' ? kex80 : kex66
```

Comment:
I think we should ensure that all versions 15.x > 15.2 should use kex80

Reply:
sure, updated below.